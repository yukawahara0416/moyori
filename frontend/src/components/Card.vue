<template>
  <v-row
    align="stretch"
    class="ma-0"
    justify="space-around"
    style="background-color: white;"
  >
    <v-col
      class="mt-2"
      cols="12"
      sm="6"
      lg="4"
      v-for="(s, id) in spots"
      :key="id"
    >
      <v-card
        class="mt-2 d-flex flex-column"
        height="100%"
        hover
        :class="{ selected: cache.id === id }"
        :id="id"
        @click.stop="setCurrentMarker(s.marker, id)"
      >
        <v-row class="pa-0">
          <v-col class="py-0" cols="5" md="12">
            <v-img
              class="white--text align-end"
              :class="{
                imgHeight_big: $vuetify.breakpoint.mdAndUp,
                imgHeight_small: $vuetify.breakpoint.smAndDown
              }"
              :src="
                s.marker.photos
                  ? s.marker.photos[0].getUrl()
                  : require('@/assets/noimage.png')
              "
            />
            <!-- :src="s.marker.photos[0].getUrl()" -->
          </v-col>

          <v-col class="py-0" cols="7" md="12">
            <v-card-text class="text--primary">
              <p>{{ s.marker.name }}</p>
              <p>{{ s.marker.address }}</p>
              <p>{{ s.marker.phone }}</p>
              <p>{{ s.marker.website }}</p>
              <!-- <p>
                <a
                  :href="
                    'https://www.google.com/maps/dir/?api=1&destination=' +
                      s.marker.position.lat +
                      ',' +
                      s.marker.position.lng
                  "
                  target="_brank"
                >
                  Googleマップでナビ
                </a>
              </p>
              <p>
                <a
                  :href="
                    'https://www.google.com/maps/search/?api=1&query=' +
                      s.marker.position.lat +
                      ',' +
                      s.marker.position.lng +
                      '&query_place_id=' +
                      s.marker.place_id
                  "
                  target="_brank"
                >
                  Googleマップで詳細を表示
                </a>
              </p> -->
            </v-card-text>
            <v-card-actions>
              <like-button :spot="s" :id="id" />
              <wifi-with-button :spot="s" :id="id" />
              <wifi-without-button :spot="s" :id="id" />
              <power-with-button :spot="s" :id="id" />
              <power-without-button :spot="s" :id="id" />
              <!-- <comment-form :spot="s" :id="id" /> -->
              <comment-button :spot="s" :id="id" />
              <spot-form :spot="s" :id="id" />
            </v-card-actions>
          </v-col>
        </v-row>
      </v-card>
    </v-col>

    <!-- カード配置対策（中央寄せ && 最下層端数左寄せ） -->
    <v-col
      class="ma-0 pa-0"
      cols="12"
      sm="6"
      lg="4"
      v-for="empty in empties"
      :key="empty"
    >
      <v-card class="flex-empty" />
    </v-col>
  </v-row>
</template>

<script>
import { mapGetters } from 'vuex'
import LikeButton from '@/components/LikeButton.vue'
import WifiWithButton from '@/components/WifiWithButton.vue'
import WifiWithoutButton from '@/components/WifiWithoutButton.vue'
import PowerWithButton from '@/components/PowerWithButton.vue'
import PowerWithoutButton from '@/components/PowerWithoutButton.vue'
// import CommentForm from '@/components/CommentForm.vue'
import CommentButton from '@/components/CommentButton.vue'
import SpotForm from '@/components/SpotForm.vue'

export default {
  components: {
    LikeButton,
    WifiWithButton,
    WifiWithoutButton,
    PowerWithButton,
    PowerWithoutButton,
    // CommentForm,
    CommentButton,
    SpotForm
  },

  computed: {
    ...mapGetters(['spots', 'cache']),

    // photoUrl(marker) {
    //   console.log(marker)
    //   if (marker.photos[0]) {
    //     return marker.photos[0].getUrl()
    //   } else {
    //     return require('@/assets/noimage.png')
    //   }
    // },

    empties() {
      return Array.from(new Array(this.spots.length))
    }
  },

  methods: {
    setCurrentMarker(marker, id) {
      this.$store.dispatch('setCurrentMarker', { marker: marker, id: id })
    }
  }
}
</script>

<style scoped>
.flex-empty {
  height: 0 !important;
  padding-top: 0 !important;
  padding-bottom: 0 !important;
  width: 230px !important;
}
.imgHeight_big {
  height: 150px !important;
}
.imgHeight_small {
  height: 100% !important;
}
.selected {
  background-color: #b3e5fc !important;
}
</style>
