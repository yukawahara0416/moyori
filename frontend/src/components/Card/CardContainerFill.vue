<template>
  <v-card class="flex-empty" />
</template>

<style scoped>
.flex-empty {
  height: 0 !important;
  padding-top: 0 !important;
  padding-bottom: 0 !important;
  width: 230px !important;
}
</style>
