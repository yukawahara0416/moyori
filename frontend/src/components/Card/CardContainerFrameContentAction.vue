<template>
  <div>
    <v-card-actions>
      <v-spacer />
      <like-button :spot="spot" :type="type" />
      <wifi-with-button :spot="spot" :type="type" />
      <wifi-without-button :spot="spot" :type="type" />
      <power-with-button :spot="spot" :type="type" />
      <power-without-button :spot="spot" :type="type" />
      <!-- <comment-button :spot="spot" :id="id" :type="type" /> -->
      <v-spacer />
    </v-card-actions>
    <v-card-actions>
      <show-button :spot="spot" :type="type" />
    </v-card-actions>
  </div>
</template>

<script>
import LikeButton from '@/components/Buttons/LikeButton.vue'
import WifiWithButton from '@/components/Buttons/WifiWithButton.vue'
import WifiWithoutButton from '@/components/Buttons/WifiWithoutButton.vue'
import PowerWithButton from '@/components/Buttons/PowerWithButton.vue'
import PowerWithoutButton from '@/components/Buttons/PowerWithoutButton.vue'
import ShowButton from '@/components/Buttons/ShowButton.vue'

export default {
  props: {
    spot: Object,
    type: String
  },

  components: {
    LikeButton,
    WifiWithButton,
    WifiWithoutButton,
    PowerWithButton,
    PowerWithoutButton,
    ShowButton
  }
}
</script>
