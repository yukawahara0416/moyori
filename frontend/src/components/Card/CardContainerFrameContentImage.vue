<template>
  <v-col class="py-0" cols="5" md="12">
    <v-img
      class="white--text align-end"
      :class="{
        imgHeight_big: $vuetify.breakpoint.mdAndUp,
        imgHeight_small: $vuetify.breakpoint.smAndDown
      }"
      :src="
        spot.marker.image ? spot.marker.image : require('@/assets/noimage.png')
      "
    />
  </v-col>
</template>

<script>
export default {
  props: {
    spot: Object
  }
}
</script>

<style scoped>
.imgHeight_big {
  height: 150px !important;
}
.imgHeight_small {
  height: 100% !important;
}
</style>
