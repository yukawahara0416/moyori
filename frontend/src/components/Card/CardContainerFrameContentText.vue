<template>
  <v-card-text class="text--primary">
    <p>{{ spot.marker.name }}</p>
    <p>{{ spot.marker.address }}</p>
    <p>{{ spot.marker.place_id }}</p>
    <p v-show="isPosted" style="color: red;">posted</p>

    <p>
      <a
        :href="
          'https://www.google.com/maps/dir/?api=1&destination=' +
            spot.data.lat +
            ',' +
            spot.data.lng
        "
        target="_brank"
      >
        Googleマップでナビ
      </a>
    </p>
  </v-card-text>
</template>

<script>
export default {
  props: {
    spot: Object
  },

  computed: {
    isPosted() {
      return 'id' in this.spot.data
    }
  }
}
</script>
