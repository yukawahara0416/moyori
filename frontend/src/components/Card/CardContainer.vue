<template>
  <v-row align="stretch" class="mt-0 mb-3 ml-0 mr-3" justify="space-around">
    <v-col
      class="pb-0 pr-0"
      cols="12"
      sm="6"
      :md="md"
      :lg="lg"
      v-for="(spot, id) in spots"
      :key="id"
    >
      <card-frame :spot="spot" :id="id" />
    </v-col>

    <!-- カード配置調整（中央寄せ && 最下層端数左寄せ） -->
    <v-col
      class="ma-0 pa-0"
      cols="12"
      sm="6"
      :md="md"
      :lg="lg"
      v-for="e in empties"
      :key="e"
    >
      <card-container-fill />
    </v-col>
  </v-row>
</template>

<script>
import CardFrame from '@/components/Card/CardFrame.vue'
import CardContainerFill from '@/components/Card/CardContainerFill.vue'

export default {
  props: {
    spots: {
      type: Array,
      default: () => {
        return []
      },
      required: true
    }
  },

  components: {
    CardFrame,
    CardContainerFill
  },

  computed: {
    empties() {
      const number = this.spots.length === 1 ? 2 : this.spots.length
      return Array.from(new Array(number))
    },

    lg() {
      const route = this.$route.name

      if (route == 'profile') return 3
      return 4
    },

    md() {
      const route = this.$route.name
      if (route == 'profile') return 4
      return 6
    }
  }
}
</script>
