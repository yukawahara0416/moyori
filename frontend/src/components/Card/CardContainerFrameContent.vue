<template>
  <v-row class="pa-0">
    <card-container-frame-content-image :spot="spot" />

    <v-col class="py-0" cols="7" md="12">
      <card-container-frame-content-title :spot="spot" />
      <card-container-frame-content-text :spot="spot" />

      <v-divider class="mx-4"></v-divider>

      <card-container-frame-content-action :spot="spot" />
    </v-col>
  </v-row>
</template>

<script>
import CardContainerFrameContentImage from '@/components/Card/CardContainerFrameContentImage.vue'
import CardContainerFrameContentTitle from '@/components/Card/CardContainerFrameContentTitle.vue'
import CardContainerFrameContentText from '@/components/Card/CardContainerFrameContentText.vue'
import CardContainerFrameContentAction from '@/components/Card/CardContainerFrameContentAction.vue'

export default {
  props: {
    spot: Object
  },

  components: {
    CardContainerFrameContentImage,
    CardContainerFrameContentTitle,
    CardContainerFrameContentText,
    CardContainerFrameContentAction
  }
}
</script>
