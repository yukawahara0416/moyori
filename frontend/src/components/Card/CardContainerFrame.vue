<template>
  <v-card
    class="mt-2 d-flex flex-column"
    height="100%"
    hover
    :class="{ selected: spot.marker.on === true }"
    :id="id"
    @click="
      spotlight(spot)
      panTo(spot)
    "
  >
    <card-container-frame-content :spot="spot" :id="id" :type="type" />
  </v-card>
</template>

<script>
import CardContainerFrameContent from '@/components/Card/CardContainerFrameContent.vue'

export default {
  props: {
    spot: Object,
    id: Number,
    type: String
  },

  components: {
    CardContainerFrameContent
  },

  methods: {
    spotlight(spot) {
      this.type === 'map'
        ? this.$store.dispatch('spot/spotlight', spot)
        : this.$store.dispatch('user/spotlight', {
            spot: spot,
            type: this.type
          })
    },

    // SearchMapWrap/mounted
    panTo(spot) {
      this.type === 'map' ? this.$root.$emit('panTo', spot) : []
    }
  }
}
</script>

<style scoped>
.selected {
  background-color: #b3e5fc !important;
}
</style>
