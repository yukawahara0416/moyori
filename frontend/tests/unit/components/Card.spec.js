import { mount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import Card from '@/components/Card.vue'
import markerStore from '@/store/modules/marker.js'

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(Vuetify)

let wrapper
let store
let state
let actions
let vuetify

beforeEach(() => {
  state = {
    spots: [
      { marker: { name: 'hoge', icon: 'foo' } },
      { marker: { name: 'fuga', icon: 'bar' } }
    ],
    cache: { id: 11, icon: 'puge' }
  }
  actions = {
    setCurrentMarker: jest.fn(),
    postSpot: jest.fn()
  }

  store = new Vuex.Store({
    modules: {
      markerStore: {
        state,
        getters: markerStore.getters,
        actions
      }
    }
  })

  vuetify = new Vuetify()

  wrapper = mount(Card, {
    localVue,
    store,
    vuetify
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe('v-on', () => {
  it('setCurrentMarker', () => {
    const event = jest.fn()
    wrapper.setMethods({ setCurrentMarker: event })
    wrapper
      .findAll('.v-card')
      .at(0)
      .trigger('click')
    expect(event).toHaveBeenCalledTimes(1)
  })

  it('postSpot', () => {
    const event = jest.fn()
    wrapper.setMethods({ postSpot: event })
    wrapper
      .findAll('[data-test="postspot"]')
      .at(0)
      .trigger('click')
    expect(event).toHaveBeenCalledTimes(1)
  })
})

describe('getters', () => {
  it('spots', () => {
    expect(wrapper.vm.spots).toEqual(state.spots)
  })

  it('cache', () => {
    expect(wrapper.vm.cache).toEqual(state.cache)
  })
})

describe('actions', () => {
  it('setCurrentMarker', () => {
    wrapper.vm.setCurrentMarker()
    expect(actions.setCurrentMarker).toHaveBeenCalled()
  })

  it('postSpot', () => {
    wrapper.vm.postSpot()
    expect(actions.postSpot).toHaveBeenCalled()
  })
})

describe('template', () => {
  it('v-for', () => {
    expect(wrapper.findAll('.v-card').length).toBe(2)
  })
  it('snapshot', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
