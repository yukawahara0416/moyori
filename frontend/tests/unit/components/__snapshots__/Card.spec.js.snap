// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template snapshot 1`] = `
<div>
  <h1>
    this is Card
  </h1>
   
  <div
    class="row"
  >
    <div
      class="col"
    >
      <div
        class="v-card v-card--hover v-card--link v-sheet theme--light"
        id="0"
        tabindex="0"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <div
              class="v-responsive v-image"
            >
              <div
                class="v-responsive__content"
              />
            </div>
          </div>
           
          <div
            class="col"
          >
            <div
              class="v-card__text"
            >
              <p>
                hoge
              </p>
               
              <p>
                
              </p>
            </div>
             
            <div
              class="v-card__actions"
            >
              <button
                class="v-btn v-btn--contained theme--light v-size--default"
                data-test="postspot"
                type="button"
              >
                <span
                  class="v-btn__content"
                >
                  
                保存
              
                </span>
              </button>
               
              <div>
                <button
                  class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"
                  type="button"
                >
                  <span
                    class="v-btn__content"
                  >
                    
    いいね
  
                  </span>
                </button>
              </div>
               
              <div>
                <button
                  class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"
                  type="button"
                >
                  <span
                    class="v-btn__content"
                  >
                    
    wifi
  
                  </span>
                </button>
              </div>
               
              <div>
                <button
                  class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"
                  type="button"
                >
                  <span
                    class="v-btn__content"
                  >
                    
    wifinone
  
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="v-card v-card--hover v-card--link v-sheet theme--light"
        id="1"
        tabindex="0"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <div
              class="v-responsive v-image"
            >
              <div
                class="v-responsive__content"
              />
            </div>
          </div>
           
          <div
            class="col"
          >
            <div
              class="v-card__text"
            >
              <p>
                fuga
              </p>
               
              <p>
                
              </p>
            </div>
             
            <div
              class="v-card__actions"
            >
              <button
                class="v-btn v-btn--contained theme--light v-size--default"
                data-test="postspot"
                type="button"
              >
                <span
                  class="v-btn__content"
                >
                  
                保存
              
                </span>
              </button>
               
              <div>
                <button
                  class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"
                  type="button"
                >
                  <span
                    class="v-btn__content"
                  >
                    
    いいね
  
                  </span>
                </button>
              </div>
               
              <div>
                <button
                  class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"
                  type="button"
                >
                  <span
                    class="v-btn__content"
                  >
                    
    wifi
  
                  </span>
                </button>
              </div>
               
              <div>
                <button
                  class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"
                  type="button"
                >
                  <span
                    class="v-btn__content"
                  >
                    
    wifinone
  
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
